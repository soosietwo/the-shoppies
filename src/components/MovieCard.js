import React from "react";
import {
  Heading,
  Subheading,
  Button,
  ResourceItem,
  TextContainer,
} from "@shopify/polaris";
import { gql, useMutation } from "@apollo/client";

import { NOMINEES_QUERY } from "./Sidebar";

const ADD_NOMINEE_MUTATION = gql`
  mutation ADD_NOMINEE_MUTATION(
    $id: ID!
    $title: String!
    $poster: String!
    $year: Int!
  ) {
    addNominee(id: $id, title: $title, poster: $poster, year: $year) {
      id
      title
      poster
      year
    }
  }
`;

const MovieCard = (props) => {
  const {
    disableAction,
    actionText,
    action,
    movie: { title, year, poster, id },
  } = props;

  const posterMarkup = (
    <img
      style={{ maxWidth: "100px" }}
      src={poster}
      alt={`Poster for ${title}`}
    />
  );

  const [
    addNominee,
    { loading, data, error },
  ] = useMutation(ADD_NOMINEE_MUTATION, {
    refetchQueries: [{ query: NOMINEES_QUERY }],
  });

  return (
    <ResourceItem
      id={id}
      media={posterMarkup}
      accessibilityLabel={`View details for ${title}`}
    >
      <TextContainer>
        <Heading>{title}</Heading>
        <Subheading>{year}</Subheading>

        {action && (
          <Button
            disabled={disableAction && disableAction(props.movie)}
            onClick={() => {
              const variables = {
                id,
                poster,
                year,
                title,
              };

              addNominee({ variables });
            }}
          >
            {actionText}
          </Button>
        )}
      </TextContainer>
    </ResourceItem>
  );
};

export default MovieCard;
