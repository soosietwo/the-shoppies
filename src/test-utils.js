// test-utils.js
import React from "react";
import { render } from "@testing-library/react";
import { AppProvider, Frame } from "@shopify/polaris";
import en from "@shopify/polaris/locales/en.json";
import Store from "./store";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const AllTheProviders = ({ children }) => {
  return (
    <AppProvider i18n={en}>
      <Store>
        <Frame>{children}</Frame>
      </Store>
    </AppProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
