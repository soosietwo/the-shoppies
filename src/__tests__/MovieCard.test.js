import React from "react";
import { render } from "../test-utils";
import { MockedProvider } from "@apollo/client/testing";
import MovieCard from "../components/MovieCard";

const mockMovie = {
  title: "test",
  year: 1234,
  poster: "123",
  id: 132,
};

describe("MovieCard", () => {
  it("renders and matches snapshot", () => {
    const { container } = render(
      <MockedProvider>
        <MovieCard movie={mockMovie} nominees={[]} />
      </MockedProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
