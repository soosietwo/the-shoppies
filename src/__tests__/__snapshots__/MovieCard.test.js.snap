// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders and matches snapshot 1`] = `
<div>
  <div>
    <div
      class="Polaris-ResourceItem"
    >
      <button
        aria-label="View details for test"
        class="Polaris-ResourceItem__Button"
        tabindex="0"
      />
      <div
        class="Polaris-ResourceItem__Container Polaris-ResourceItem--alignmentCenter"
        id="132"
      >
        <div
          class="Polaris-ResourceItem__Owned"
        >
          <div
            class="Polaris-ResourceItem__Media"
          >
            <img
              alt="Poster for test"
              src="123"
              style="max-width: 100px;"
            />
          </div>
        </div>
        <div
          class="Polaris-ResourceItem__Content"
        >
          <div
            class="Polaris-TextContainer"
          >
            <h2
              class="Polaris-Heading"
            >
              test
            </h2>
            <h3
              class="Polaris-Subheading"
            >
              1234
            </h3>
            <button
              class="Polaris-Button"
              type="button"
            >
              <span
                class="Polaris-Button__Content"
              >
                <span
                  class="Polaris-Button__Text"
                >
                  Add nominee
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
